hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/bucket/OMBucketCreateRequest.java

```diff
@@ -145,12 +145,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
     OmBucketInfo omBucketInfo = null;
     if (bucketInfo.getBucketLayout() == null || bucketInfo.getBucketLayout()
         .equals(BucketLayoutProto.LEGACY)) {
-      BucketLayout defaultType = BucketLayout.LEGACY;
-      if (StringUtils
-          .equalsIgnoreCase(OMConfigKeys.OZONE_OM_METADATA_LAYOUT_PREFIX,
-              omLayout)) {
-        defaultType = BucketLayout.FILE_SYSTEM_OPTIMIZED;
-      }
+      BucketLayout defaultType = OMConfigKeys.OZONE_BUCKET_LAYOUT_DEFAULT;
```

Comment:
Please read the value from OMConfiguration instead of reading a static value. 

Below is sample code reference. 
[OMFileCreateRequest.java#L90](https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileCreateRequest.java#L90)

In your patch, you can do `ozoneMgr.getConfiguration().getTrimmed(OMConfigKeys.OZONE_DEFAULT_BUCKET_LAYOUT OMConfigKeys.OZONE_BUCKET_LAYOUT_DEFAULT);`

Reply:
@JyotinderSingh  Please include test case to verify the behavior.

Reply:
Thanks for the review, working on this.